### PR TITLE
Support CPTV v2 files

### PR DESCRIPTION
**Pull request by mjs**
- Handle the new v2 fields
- Expose the CPTV file number
- Frames are now returned as a slots based (i.e. fast) objects
- Include example script which reads a file and prints out contents
- Link to v2 SPEC in README

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -4,7 +4,7 @@ This is Python package provides for quick, easy parsing for Cacophony
 Project Thermal Video (CPTV) files. It works with **Python 3** only.
 
 For more details on the internals of CPTV files, see the
-[specification](https://github.com/TheCacophonyProject/go-cptv/blob/master/SPEC.md).
+[specification](https://github.com/TheCacophonyProject/go-cptv/blob/master/SPECv2.md).
```
Comment (clare) [inline README.md:7]: This also supports v1 at the moment and we need it too (else machine learning will be hard....so we should put that here???

File: README.md [hunk 2]
```diff
@@ -28,15 +28,25 @@ from cptv import CPTVReader
...
-    print(reader.device_name)
-    print(reader.timestamp)
-    print(reader.x_resolution)
-    print(reader.y_resolution)
-
-    for frame, t_offset in reader:
-        # Do something with frame.
-        # Each frame is a 2D numpy array.
-
+    print("version:", reader.version)
```
Comment (clare) [inline README.md]: This is now duplicated with read.py.   Would a link be  better?

**Review by clare — APPROVED** [review]
Lots that could be tested here with automated tests.   Important for maintaining backward compatibility.

File: README.md (hunk 2 above)
Comment (mjs) [reply README.md]: Good thinking. Fixed.

**Pull request merged**